modules/chempy/cpv.py

```diff
@@ -68,92 +71,85 @@ def random_displacement(v,radius):
-def cross_product(v1,v2):
+def cross_product(v1: Sequence[float], v2: Sequence[float]) -> list[float]:
   return [(v1[1]*v2[2]) - (v1[2]*v2[1]),
              (v1[2]*v2[0]) - (v1[0]*v2[2]),
              (v1[0]*v2[1]) - (v1[1]*v2[0])]
 
 #------------------------------------------------------------------------------
-def transform(m,v):
+def transform(m: Sequence[Sequence[float]], v: Sequence[float]) -> list[float]:
     return [m[0][0]*v[0] + m[0][1]*v[1] + m[0][2]*v[2],
               m[1][0]*v[0] + m[1][1]*v[1] + m[1][2]*v[2],
               m[2][0]*v[0] + m[2][1]*v[1] + m[2][2]*v[2]]
 
 #------------------------------------------------------------------------------
-def inverse_transform(m,v):
+def inverse_transform(m: Sequence[Sequence[float]], v: Sequence[float]) -> list[float]:
     return [m[0][0]*v[0] + m[1][0]*v[1] + m[2][0]*v[2],
             m[0][1]*v[0] + m[1][1]*v[1] + m[2][1]*v[2],
             m[0][2]*v[0] + m[1][2]*v[1] + m[2][2]*v[2]]
 
 #------------------------------------------------------------------------------
-def multiply(m1,m2): # HAVEN'T YET VERIFIED THAT THIS CONFORMS TO STANDARD DEFT
-    return [[m1[0][0]*m2[0][0] + m1[0][1]*m2[1][0] + m1[0][2]*m2[2][0],
-                m1[1][0]*m2[0][0] + m1[1][1]*m2[1][0] + m1[1][2]*m2[2][0],
-                m1[2][0]*m2[0][0] + m1[2][1]*m2[1][0] + m1[2][2]*m2[2][0]],
-              [m1[0][0]*m2[0][1] + m1[0][1]*m2[1][1] + m1[0][2]*m2[2][1],
-                m1[1][0]*m2[0][1] + m1[1][1]*m2[1][1] + m1[1][2]*m2[2][1],
-                m1[2][0]*m2[0][1] + m1[2][1]*m2[1][1] + m1[2][2]*m2[2][1]],
-              [m1[0][0]*m2[0][2] + m1[0][1]*m2[1][2] + m1[0][2]*m2[2][2],
-                m1[1][0]*m2[0][2] + m1[1][1]*m2[1][2] + m1[1][2]*m2[2][2],
-                m1[2][0]*m2[0][2] + m1[2][1]*m2[1][2] + m1[2][2]*m2[2][2]]]
+def multiply(m1: Sequence[Sequence[float]], m2: Sequence[Sequence[float]]) -> list[list[float]]:
+   # HAVEN'T YET VERIFIED THAT THIS CONFORMS TO STANDARD DEFT
+   # upd: no, it's not(fixed)
 
-#------------------------------------------------------------------------------
-def transpose(m1):
-    return [[m1[0][0],
-             m1[1][0],
-             m1[2][0]],
-            [m1[0][1],
-             m1[1][1],
-             m1[2][1]],
-            [m1[0][2],
-             m1[1][2],
-             m1[2][2]]]
+   return [[m1[0][0]*m2[0][0] + m1[0][1]*m2[1][0] + m1[0][2]*m2[2][0],
+               m1[0][0]*m2[0][1] + m1[0][1]*m2[1][1] + m1[0][2]*m2[2][1],
+               m1[0][0]*m2[0][2] + m1[0][1]*m2[1][2] + m1[0][2]*m2[2][2]],
+             [m1[1][0]*m2[0][0] + m1[1][1]*m2[1][0] + m1[1][2]*m2[2][0],
+               m1[1][0]*m2[0][1] + m1[1][1]*m2[1][1] + m1[1][2]*m2[2][1],
+               m1[1][0]*m2[0][2] + m1[1][1]*m2[1][2] + m1[1][2]*m2[2][2]],
+             [m1[2][0]*m2[0][0] + m1[2][1]*m2[1][0] + m1[2][2]*m2[2][0],
+               m1[2][0]*m2[0][1] + m1[2][1]*m2[1][1] + m1[2][2]*m2[2][1],
+               m1[2][0]*m2[0][2] + m1[2][1]*m2[1][2] + m1[2][2]*m2[2][2]]]
```

Comment:
Matrix multiplication was not implemented correctly. This and code duplication has also been fixed in this PR